File: mesop/components/card_content/card_content.py
import mesop.components.card_content.card_content_pb2 as card_content_pb
from mesop.component_helpers import (
  insert_composite_component,
  register_native_component,
)


@register_native_component
def card_content(
  *,
  key: str | None = None,
):
  """
  This function creates a card_content.

  This component is meant to be used with the `card` component. It is used for the
  contents of a card that

  This component is a optional. It is mainly used as a convenience for consistent
  formatting with the card component.
  """
  return insert_composite_component(
    key=key,
    type_name="card_content",
    proto=card_content_pb.CardContentType(),
  )

File: mesop/components/card_content/card_content.ts
import {MatCardModule} from '@angular/material/card';
import {Component, Input} from '@angular/core';
import {
  Key,
  Type,
} from 'mesop/mesop/protos/ui_jspb_proto_pb/mesop/protos/ui_pb';
import {CardContentType} from 'mesop/mesop/components/card_content/card_content_jspb_proto_pb/mesop/components/card_content/card_content_pb';

@Component({
  selector: 'mesop-card-content',
  templateUrl: 'card_content.ng.html',
  standalone: true,
  imports: [MatCardModule],
})
export class CardContentComponent {
  @Input({required: true}) type!: Type;
  @Input() key!: Key;
  private _config!: CardContentType;

  ngOnChanges() {
    this._config = CardContentType.deserializeBinary(
      this.type.getValue() as unknown as Uint8Array,
    );
  }

  config(): CardContentType {
    return this._config;
  }
}
